### PR TITLE
datagen changes for perf tests

### PR DESCRIPTION
**Pull request by rodesai**
- Support using multiple producer threads
- Toggle printing records to console
- Use a thread-local generator for producing random strings
- Support generating random strings with a fixed domain. A domain restricts the
  number of possible generated strings. Used to control key-space.
- Some configs to control timestamp generation
- Allow specifying producer configs at the command-line

### CODE REVIEW

**Comment by apurvam** [comment]
bump! @rodesai probably worth rebasing this onto master.

**Comment by rodesai** [comment]
WIll do. I have to add in the rate limiting stuff as well. I'll mark it WIP until its ready for review again.

File: ksql-examples/src/main/java/io/confluent/avro/random/generator/Generator.java [hunk 1]
```diff
@@ -1130,16 +1137,33 @@ private String generateRegexString(Schema schema, Object regexProp, LengthBounds
 
   private String generateRandomString(int length) {
     byte[] bytes = new byte[length];
+    ThreadLocalRandom tlRandom = ThreadLocalRandom.current();
     for (int i = 0; i < length; i++) {
-      bytes[i] = (byte) random.nextInt(128);
+      bytes[i] = (byte)((byte)'a' + ((byte) tlRandom.nextInt(32) % 26));
```
Comment (C0urante) [inline ksql-examples/src/main/java/io/confluent/avro/random/generator/Generator.java]: Monsanto had a [less-restrictive approach](https://github.com/MonsantoCo/avro-random-generator/commit/63f96001e0a76818c8b33b2a3e33670d905d8a33) for this change that I think might make more sense here. If someone wants only lowercase characters in a string, they can use a regex like `[a-z]*`.

Comment (rodesai) [reply ksql-examples/src/main/java/io/confluent/avro/random/generator/Generator.java]: I had reasons for why this is so weird. I can't remember now. Something performance-related. Let met think about it.

Comment (C0urante) [reply ksql-examples/src/main/java/io/confluent/avro/random/generator/Generator.java]: Looks like calling `ThreadLocalRandom.nextInt(...)` with a power of 2 has good performance implications. Performing modulus on the resulting value with a number that isn't a power of 2 causes skewed results, though.

Sample program:
```
import java.util.concurrent.ThreadLocalRandom;
import java.util.Map;
import java.util.HashMap;

public class Test {
    public static void main(String[] args) throws Exception {
        Map<Integer, Integer> occurrences = new HashMap<>();
        for (int i = 0; i < 1024 * 1024; i++) {
            occurrences.compute(
                ThreadLocalRandom.current().nextInt(32) % 26,
                (k, v) -> v != null ? v + 1 : 1
            );
        }
        System.out.println(occurrences);
    }
}
```

Running that once gave output of:
```
{0=66056, 1=65473, 2=65435, 3=65994, 4=65377, 5=65510, 6=32973, 7=32877, 8=32608, 9=32791, 10=32425, 11=32798, 12=32976, 13=32747, 14=32622, 15=32761, 16=32703, 17=32626, 18=32780, 19=33085, 20=32870, 21=32391, 22=32761, 23=32554, 24=32867, 25=32516}
```

**Pull request closed without merging**